### ISSUE TITLE
host - container path not found issue

### ISSUE THREAD

**Issue by etranger05** [AntelopeIO/DUNE#10]
## Overview
Found 'path not found' internal error after ```dune --create-cmake-app``` cmd run.

This issue had also reported at the previous mandel DUNE repository but it looks not resolved yet.
https://github.com/eosnetworkfoundation/DUNE/issues/21

## Environment
macOS 12.5 (M1 Apple Silicon)
DUNE v1.0.0.rc2
Docker Desktop 4.11.1 (84025)
Docker Engine 20.10.17
Python 3.9.13
SHELL: zsh

## Reproduce steps
Install Docker & Python3, clone the DUNE repo, set the PATH, run bootstrap.sh as described in README.md.
And created a directory for contracts under the DUNE path, and run the ```dune --create-cmake-app ``` command as below.
```
$ mkdir contracts
$ cd contracts
$ dune --create-cmake-app hello ./
/host/Users/junhank/Documents/workspace/AntelopeDUNE/contracts path not found

Internal Error

$ dune --version
DUNE v1.0.0.rc2
```
I'm not sure it only happens in macOS as my Ubuntu machine for test has passed away. 
Anyway, there are 2 workarounds I found for this issue.

## Workaround -1
Edit docker.py file before start dune container instance as below.

1. Open below
```
<PathToDUNE>/DUNE/src/dune/docker.py
```
2. Edit return path under the ```abs_host_path``` as below and save.
```
def abs_host_path(self, dir):
      ...
      return '/host/host_mnt/'+abs_path
```
3. Start dune instance
```
dune --start <node_name>
```
4. Run ```dune --create-cmake-app ``` cmd.

You may need to grant Disk Access to docker from System Preference.
I didn't test all the features of DUNE, but at least I have successfully built and deployed a sample smart contract after these steps.

## Workaround -2
1. Before run ./bootstrap.sh, Launch Docker Desktop → settings

![image](https://user-images.githubusercontent.com/11245288/185585521-2584a200-24c9-46ab-8d92-db1617132e9f.png)

2. Uncheck “use gRPC FUSE for file sharing” to use the legacy osxfs.
3. Apply & Restart
4. Run ./bootstrap.sh
5. Run ```dune --create-cmake-app ``` cmd.

It will lose performance syncing between host and container, but it was ok for me to run simple smart contracts.

Thank you.

**Comment by wackou** [comment]
Same issue happened to me, it looks like it's an issue with mounting the root folder from the host. In the `/host` folder inside the container there are links that seem broken and out of place, and it only happens when mounting `/`, not with other folders that I tried

I found a fix that doesn't look like a hack, but I'm a docker noob so I'm not sure it's the proper one... :)

I went to Docker desktop `Settings > Resources > File Sharing` and I added `/` as a folder that can be bind mounted. Restart and then it works for me. Hope that helps.
